src/DependencyInjection/Configuration.php

```diff
@@ -214,9 +214,13 @@ private function addCacheControlSection(ArrayNodeDefinition $rootNode)
                             ->scalarNode('stale_while_revalidate')->end()
                         ->end()
                     ->end()
-                    ->booleanNode('etag')
+                    ->enumNode('etag')
                         ->defaultValue(false)
-                        ->info('Set a simple ETag which is just the md5 hash of the response body')
+                        ->treatTrueLike('strong')
+                        ->info('Set a simple ETag which is just the md5 hash of the response body. '.
+                               'You can specify which type of ETag you want by passing "strong" or "weak", '.
+                               'value true is treated as "strong".')
+                        ->values(['weak', 'strong', false, true])
```

Comment:
please remove true here, the treatTrueLike should take care to always translate this to "strong".